libraries-stream/src/test/java/com/baeldung/parallel_collectors/ParallelCollectorsVirtualThreadsManualTest.java

```diff
@@ -0,0 +1,69 @@
+package com.baeldung.parallel_collectors;
+
+import com.pivovarit.collectors.ParallelCollectors;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toList;
+
+public class ParallelCollectorsVirtualThreadsManualTest {
+
+    @Test
+    public void processInParallelOnOSThreads() {
+        int parallelProcesses = 5_000;
+
+        var e = Executors.newFixedThreadPool(parallelProcesses);
+
+        var result = timed(() -> Stream.iterate(0, i -> i + 1).limit(parallelProcesses)
+          .collect(ParallelCollectors.parallel(i -> fetchById(i), toList(), e, parallelProcesses))
+          .join());
+
+        System.out.println(result);
+    }
+
+    @Test
+    public void processInParallelOnVirtualThreads() {
```

Comment:
Could we update the test names to comply with the BDD convention: givenX_whenY_thenZ?

Reply:
Done